libs/@guardian/libs/src/consent-management-platform/vendorDataManager.ts

```diff
@@ -3,31 +3,42 @@ import { storage } from '../storage/storage';
 import { getConsentFor } from './getConsentFor';
 import { onConsentChange } from './onConsentChange';
 import type { ConsentState } from './types';
-import type { VendorWithData } from './vendorStorageIds';
-import { vendorStorageIds } from './vendorStorageIds';
+import type { VendorData, VendorWithData } from './vendorStorageIds';
+import {
+	deprecatedVendorStorageIds,
+	vendorStorageIds,
+} from './vendorStorageIds';
 
-const removeData = (consent: ConsentState) =>
+const removeData = (vendorData: VendorData): void => {
```

Comment:
By using destructuring here and setting defaults you can safely loop over each array rather than having to check for their existence first:

```suggestion
const removeData = ({
  cookies = [],
  localStorage = [],
  sessionStorage = [],
}: VendorData): void => {
```

Reply:
Implemented in #1569